File: kattis/easy/aboveaverage.java
package easy;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class aboveaverage {
	private static double calcAvg(StringTokenizer tokenizer, int students, int[] grades) {
		double total = 0;
		for (int j = 0; j < students; j++) {
			grades[j] = Integer.parseInt(tokenizer.nextToken());
			total += grades[j];
		}

		// average calculation
		return (students == 0) ? 0 : total / students;
	}

	private static double calcAboveAvg(int[] grades, double avgGrade) {
		double count = 0;
		for (int grade : grades)
			if (grade > avgGrade) count++;

		return count;
	}

	public static void main(String[] args) throws IOException {
		try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
			int cases = Integer.parseInt(br.readLine());

			for (int i = 0; i < cases; i++) {
				StringTokenizer tokenizer = new StringTokenizer(br.readLine());
				int students = Integer.parseInt(tokenizer.nextToken());
				int[] grades = new int[students];

				// Read grades and calculate average
				double avgGrade = calcAvg(tokenizer, students, grades);

				// Calculate percentage of students above average
				double aboveAverageCount = calcAboveAvg(grades, avgGrade);

				// Calculate and print result
				double percentageAboveAverage = (aboveAverageCount / students) * 100;
				System.out.printf("%.3f%%%n", percentageAboveAverage);
			}
		}
	}
}

File: kattis/easy/cpp/aboveaverage.cpp
//
// Created by BBJ on 31.12.2023.
//
#include <iostream>
#include <vector>
#include <iomanip>

using namespace std;
#define for_i(x) for(int i =0; i < x; i++)


double calcAvg(vector<int>& grades, int students) {
    double total = 0;
    for_i(students) {
        cin >> grades[i];
        total += grades[i];
    }
    return (students == 0) ? 0 : total / students;
}

double calcAboveAvg(vector<int>& grades, double avg) {
    double count = 0;
    for (int grade : grades)
        if (grade > avg) count++;

    return count;
}

int main() {

    int cases; cin >> cases;

    for_i(cases) {
        int students; cin >> students;
        vector<int> grades(students);

        double avGrade = calcAvg(grades, students);

        double aboveAverageCount = calcAboveAvg(grades, avGrade);

        double percentAbove = (aboveAverageCount / students) * 100;

        cout << fixed << setprecision(3) << percentAbove << "%" << endl;

    }

    return 0;
}